iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl

```diff
@@ -103,7 +103,9 @@ inline cxx::expected<ChunkDistributorError> ChunkDistributor<ChunkDistributorDat
 {
     typename MemberType_t::LockGuard_t lock(*getMembers());
 
-    const auto iter = std::find(getMembers()->m_queues.begin(), getMembers()->m_queues.end(), queueToRemove);
+    const auto iter = std::find(getMembers()->m_queues.begin(),
+                                getMembers()->m_queues.end(),
+                                static_cast<ChunkQueueData_t* const>(queueToRemove));
```

Comment:
So `not_null` does not implicitly convert to its template type? Looks more like a problem that should be addressed there then (and probably would lead to  other changes).

I do not see why the changes in `RelativePointer` imply this change though...

Reply:
We want to avoid any implicit conversion and make the conversion explicit as defined in both the AUTOSAR and the MISRA ruleset. That makes sense to me. The change is needed because `std::find` needs the comparison operator, which can use implicit conversion if implemented inside the class but not if the implementation is outside of the class. This is also the reason AUTOSAR and MISRA have this rule.